src/GitHub.App/Services/RepositoryPublishService.cs

```diff
@@ -14,21 +14,21 @@ namespace GitHub.Services
     public class RepositoryPublishService : IRepositoryPublishService
     {
         readonly IGitClient gitClient;
-        readonly IRepository activeRepository;
+        readonly IVSGitServices vsGitServices;
 
         [ImportingConstructor]
         public RepositoryPublishService(IGitClient gitClient, IVSGitServices vsGitServices)
         {
             this.gitClient = gitClient;
-            this.activeRepository = vsGitServices.GetActiveRepo();
+            this.vsGitServices = vsGitServices;
         }
 
         public string LocalRepositoryName
         {
             get
             {
-                if (!string.IsNullOrEmpty(activeRepository?.Info?.WorkingDirectory))
-                    return new DirectoryInfo(activeRepository.Info.WorkingDirectory).Name ?? "";
+                if (!string.IsNullOrEmpty(vsGitServices.GetActiveRepo()?.Info?.WorkingDirectory))
+                    return new DirectoryInfo(vsGitServices.GetActiveRepo().Info.WorkingDirectory).Name ?? "";
```

Comment:
`GetActiveRepo` is a somewhat heavy operation, so it's probably best to cache the value here to avoid the double hit on it.